app/components/FileOperationListItem.js

```diff
@@ -0,0 +1,158 @@
+// @flow
+import { LinkIcon } from "outline-icons";
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import { Link } from "react-router-dom";
+import styled, { css } from "styled-components";
+import breakpoint from "styled-components-breakpoint";
+import FileOperation from "models/FileOperation";
+import Avatar from "components/Avatar";
+import Badge from "components/Badge";
+import Button from "components/Button";
+import EventBoundary from "components/EventBoundary";
+import Flex from "components/Flex";
+import Time from "components/Time";
+import useCurrentUser from "hooks/useCurrentUser";
+import useMobile from "hooks/useMobile";
+import useQuery from "hooks/useQuery";
+
+type FileOperationListItemProps = {
+  item: FileOperation,
+};
+
+const FileOperationListItem = ({ item }: FileOperationListItemProps) => {
+  const user = useCurrentUser();
+  const { t } = useTranslation();
+  const isMobile = useMobile();
+  const params = useQuery();
+  const key = params.get("key") || "";
+
+  return (
+    <ImportExportListItemWrapper highlight={key === item.id}>
+      <Flex align="center" column>
+        <AvatarWrapper src={item.user.avatarUrl} size={32} />
+      </Flex>
+      <Content>
+        <Heading>
+          {item.collection ? (
+            <Title>
+              <Link to={item.collection.url}>{item.collection.name}</Link>
+            </Title>
+          ) : (
+            <Title>All collections</Title>
+          )}
+          <Badge>{item.state}</Badge>
+        </Heading>
+        <Container>
+          <Flex>
+            {user.id === item.user.id ? t("You") : item.user.name} created&nbsp;
+            <Time dateTime={item.createdAt} addSuffix shorten />
+          </Flex>
+          {!isMobile && <>&nbsp;•&nbsp;</>}
+          <Flex>{(item.size / (1024 * 1024)).toPrecision(2)}MB</Flex>
+        </Container>
+      </Content>
+
+      {item.url && (
+        <Actions>
+          <Button
+            onClick={() => window.open(item.url, "_blank")}
```

Comment:
The `url` is a direct link to the storage without any authentication so this isn't going to work in production setups – all of the uploaded files require a presigned url (the same as attachments).

I think we'll need to use the same approach here as attachments where the is an api url, eg `/api/file_operations.redirect?id=<uuid>` that would calculate and redirect to the presigned url on click.

This would also actually allow us to embed a link in the email which would work as long as the user is signed in – seems fine? 🤔

Reply:
Not sure if it will allow us to embed a link because we have to hit Outline API using `POST`, Isn't it?

Reply:
You can use `GET`, there is some api middleware that handles this 👍